### DISCUSSION TITLE
New App Help and Ideas

### DISCUSSION THREAD

**Discussion by gdeeble** [community-scripts/ProxmoxVE#1126] [General]
Hey all!

First thanks for keeping this stuff going. It was definitely heartbreaking hearing about Tteck, back in November, but his and these scripts have been a wonderful resource with setting up some of these apps.

I'm still very new to using Github and development processes outlined so please bear with me on this. Most of my development has been private and honestly willy nilly. I'm trying to make sure I stick to the guidelines for consistency and hopefully have done ok. I've put the install scripts for Openspeedtest in here and was wondering if anyone would be up for critiquing them, maybe giving pointers for anything that could be done better. They are pointing to my repo so I could test the installation process, and have confirmed it works with both Debian 12 and Ubuntu 24.04 without any issues.

While writing out the scripts for Openspeedtest, I notice that if I don't update the source in the CT script and the "community-scripts" reference in the function files, it wouldn't work(as expected since they don't exist here). My thought would be to see if we could set up a global variable that was set before importing the function scripts so that they could point to a private repo(mainly for creation and testing purposes) so that it could be dynamically configured at the time of execution. My thought would be if adding a way to check for a repo argument and set that in the script as the global variable, then call it everywhere else with the understanding that if it's not set then it would utilize community-scripts as the repo.

Please let me know if there is any feedback, as I look at this as an opportunity to give back to the community, not to mention an opportunity to strengthen my skills.

Scripts:
https://raw.githubusercontent.com/gdeeble/ProxmoxVE/refs/heads/main/ct/openspeedtest.sh
https://raw.githubusercontent.com/gdeeble/ProxmoxVE/refs/heads/main/install/openspeedtest-install.sh

Cheers!
-Gary

**Comment by MickLesk** [comment]
you need to edit: build.func and install.func, there are curls about this project

**Reply by gdeeble** [reply]
Hey MickLesk! 

Happy New Years Eve!

For the PR process, do I leave them as my repo so they can be tested without importing them here, or do I need to update them to point to community-scripts, so it can be pulled in and then tested? 

My suggestion of the global variable/parameter is only to make development easier on those using private repos. I can scope out the function files and see how hard it would be to change it to variables. With a fresh day, it came to mind that not only the user(what I actually meant at the time of the posting) but branch probably should be this way too.  Maybe as simple as exporting them before excluding the script and if they aren't there, they can default to the norm. If they are implemented, then people would be able to leave the default source in their script which would pull from here but would be able to update downstream with 0 impact to everything else. 

Something like this:
Pull build.func from community-scripts
build.func does a check for "GIT_USER" and "GIT_BRANCH" environment variables
if they are not set, set GIT_USER to community-scripts and GIT_BRANCH to main
all remaining calls that look at the repo use those 2 variables instead of hardcoded values

Does that kind of make sense what I was thinking? Some times I have a half baked ideas and maybe this is one of them that's not so great.

Sorry if I sound like a PITA, I'm just thinking on how can this be made easier for novice developers like myself and add scalability. Removing the necessity of editing those files would make it easier to build scripts without the necessity of finding those references.
Thanks!
-Gary

**Reply by gdeeble** [reply]
As a test to make sure my thought process works, I went ahead and added 2 if statements to the base_settings function, to set the variables, and updated most of the references where it would be needed in the function files. So if I run it without out GIT_USER and/or GIT_BRANCH set, it will default to community-scripts and main, but if I set either of them, it will use them instead. 

To see it in action you can run bash -c "$(wget -qLO - https://raw.githubusercontent.com/gdeeble/ProxmoxVE/refs/heads/main/ct/openspeedtest.sh)" and it should build an empty container as is can't get to the installer script. But is you run export GIT_USER=gdeeble before executing the bash script, it should build the container properly. 

Personal feelings are that doing this would open up the ability for new developers to assist with ease. I'm pretty sure the code is ok, but there may be a better way of handling it, but feel free to take any pieces deemed as a good idea.

Cheers!
-Gary

**Reply by kendikaku** [reply]
does this script work?
I'm looking for a script to install in lxc and so far I haven't found it.

edit:

export GIT_USER=gdeeble
bash -c "$(wget -qLO - https://raw.githubusercontent.com/gdeeble/ProxmoxVE/refs/heads/main/ct/openspeedtest.sh)"

it worked!

**Reply by gdeeble** [reply]
> does this script work? I'm looking for a script to install in lxc and so far I haven't found it.
> 
> edit:
> 
> export GIT_USER=gdeeble bash -c "$(wget -qLO - https://raw.githubusercontent.com/gdeeble/ProxmoxVE/refs/heads/main/ct/openspeedtest.sh)"
> 
> it worked!

Awesome. Please peep in mind this is from my personal repo where I was just developing a potential solution for it. OST is current, but if there are future updates, it may not take unless Mick and team pulls over scripts for it (which if they just want to take the code directly, I have no problem with that). I'm not sure the entire PR process so I haven't opened a PR at this point.
